### PR TITLE
feat: make GNS signal transferable

### PR DESCRIPTION
**Pull request by tmigone**
See https://forum.thegraph.com/t/gip-0032-make-gns-signal-transferrable/3338

### Motivation

From GIP32:

> Curators can mint curation shares from a Subgraph Deployment directly on the Curation contract and get fully transferrable ERC20 shares. However, when they curate a Subgraph at the GNS level, the GNS contract is the one keeping all the signal shares under its control. The subgraph curators cannot transfer those shares because the GNS is not exposing any function to do that.

### Changes

This PR adds a `transferSignal` function to the GNS contract that allows a curator to transfer ownership of their `nSignal` to another address.

Signed-off-by: Tomás Migone <tomas@edgeandnode.com>

### CODE REVIEW

File: contracts/discovery/GNS.sol [hunk 1]
```diff
@@ -86,6 +86,16 @@ contract GNS is GNSV2Storage, GraphUpgradeable, IGNS, Multicall {
         uint256 tokensReceived
     );
 
+    /**
+     * @dev Emitted when a curator transfers signal.
+     */
+    event SignalTransfered(
```
Comment (pcarranzav) [inline contracts/discovery/GNS.sol:92]: nit/typo: should be `SignalTransferred`

File: contracts/discovery/GNS.sol [hunk 2]
```diff
@@ -86,6 +86,16 @@ contract GNS is GNSV2Storage, GraphUpgradeable, IGNS, Multicall {
...
+        uint256 indexed subgraphID,
+        address indexed from,
+        address indexed to,
+        uint256 nSignalTransfered
```
Comment (pcarranzav) [inline contracts/discovery/GNS.sol:96]: nit/typo: should be `nSignalTransferred`

**Review by abarmat — DISMISSED** [review]
This is ready for an audit

**Comment by tmigone** [comment]
Latest commit https://github.com/graphprotocol/contracts/pull/568/commits/09f40d160b2a65e9672ca005fa85e329bd52d3e4 addresses audit issues N-22 and L-11

**Review by abarmat — APPROVED** [review]

**Pull request merged**